frontend/src/components/TwoLevelDropdown.tsx

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2022 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Ref, useRef, useState } from 'react';
+import { Button, Tooltip } from '@material-ui/core';
+import { color } from 'src/Css';
+import { classes, stylesheet } from 'typestyle';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+const css = stylesheet({
+  defaultFont: {
+    fontSize: '1rem',
+  },
+  dropdown: {
+    border: `1px solid ${color.lightGrey}`,
+    color: color.grey,
+    fontWeight: 400,
+    margin: 0,
+  },
+  dropdownMenu: {
+    display: 'block',
+    borderRadius: '5px',
+    border: `1px solid ${color.lightGrey}`,
+    margin: 0,
+    padding: '5px 0',
+    position: 'absolute',
+    top: '100%',
+    left: 0,
+    minWidth: '10rem',
+    cursor: 'pointer',
+    backgroundColor: color.background,
+  },
+  dropdownElement: {
+    padding: '1rem',
+    position: 'relative',
+
+    $nest: {
+      '&:hover': {
+        backgroundColor: `${color.lightGrey}`,
+      },
+    },
+  },
+  textContainer: {
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    maxWidth: '30rem',
+    overflow: 'hidden',
+    verticalAlign: 'top',
+  },
+  dropdownSubmenu: {
+    left: '100%',
+    // Offsets the dropdown menu vertical padding (5px) and border (1px).
+    top: '-6px',
+  },
+  inlineContainer: {
+    display: 'inline-block',
+  },
+  relativeContainer: {
+    position: 'relative',
+  },
+});
+
+const ChevronRightSmallIcon = () => <ChevronRightIcon className={classes(css.defaultFont)} />;
+
+export interface SelectedItem {
+  itemName: string;
+  subItemName: string;
+  subItemSecondaryName?: string;
+}
+
+interface DropdownSubItem {
+  name: string;
+  secondaryName?: string;
+}
```

Comment:
Is below understanding correct?

Item name => pipeline run name
subItemName => pipeline step name
sbuItemSecondaryName => Artifact name

Will different artifacts sharing same subItemName be close to each other and sorted when rendering `<li>` ?

Reply:
`subItemName => pipeline execution name`; this is required since we need the combination of execution name and artifact name in order to uniquely identify a visualization.

Yes, different artifacts sharing the same `subItemName` will be closer to each other and sorted when rendering `<li>`. The items will already be sorted when they are passed into the `items: DropdownItem[]` parameter; no sorting is done within the `TwoLevelDropdown` component.